bin/compile

```diff
@@ -1,102 +1,101 @@
 #!/bin/bash
 
-# `set –e` is used within the Bash to stop execution instantly as a query exits while having a non-zero status. 
+# `set –e` is used within Bash to stop execution instantly as a query exits with a non-zero status.
 # This function is also used when you need to know the error location in the running code
 set -e
 
 BUILD_DIR=$1
+CACHE_DIR=$2
 ENV_DIR=$3
+ROOT_DIR=$(pwd)
 
 # Get env variables
-ENV_ROLE_ARN=$(test -f "$ENV_DIR/ROLE_ARN" && cat "$ENV_DIR/ROLE_ARN" || echo "")
-ENV_S3_BUCKET_NAME=$(test -f "$ENV_DIR/S3_BUCKET_NAME" && cat "$ENV_DIR/S3_BUCKET_NAME" || echo "")
-ENV_DISTRIBUTION_ID=$(test -f "$ENV_DIR/DISTRIBUTION_ID" && cat "$ENV_DIR/DISTRIBUTION_ID" || echo "")
-ENV_AWS_ACCESS_KEY_ID=$(test -f "$ENV_DIR/AWS_ACCESS_KEY_ID" && cat "$ENV_DIR/AWS_ACCESS_KEY_ID" || echo "")
-ENV_AWS_SECRET_ACCESS_KEY=$(test -f "$ENV_DIR/AWS_SECRET_ACCESS_KEY" && cat "$ENV_DIR/AWS_SECRET_ACCESS_KEY" || echo "")
-ENV_HEROKU_APP_NAME=$(test -f "$ENV_DIR/HEROKU_APP_NAME" && cat "$ENV_DIR/HEROKU_APP_NAME" || echo "")
-ENV_HEROKU_RELEASE_VERSION=$(test -f "$ENV_DIR/HEROKU_RELEASE_VERSION" && cat "$ENV_DIR/HEROKU_RELEASE_VERSION" || date +%s )
-
-if [ -z "$ENV_ROLE_ARN" ]; then
-  echo "-----> ⏭️  Skipping AWS sync. ROLE_ARN is not set"
-  exit 0
-fi
+ENV_ROLE_ARN=$(cat "$ENV_DIR/ROLE_ARN" 2>/dev/null || true)
+ENV_S3_BUCKET_NAME=$(cat "$ENV_DIR/S3_BUCKET_NAME" 2>/dev/null || true)
+ENV_DISTRIBUTION_ID=$(cat "$ENV_DIR/DISTRIBUTION_ID" 2>/dev/null || true)
+ENV_AWS_ACCESS_KEY_ID=$(cat "$ENV_DIR/AWS_ACCESS_KEY_ID" 2>/dev/null || true)
+ENV_AWS_SECRET_ACCESS_KEY=$(cat "$ENV_DIR/AWS_SECRET_ACCESS_KEY" 2>/dev/null || true)
+ENV_HEROKU_APP_NAME=$(cat "$ENV_DIR/HEROKU_APP_NAME" 2>/dev/null || true)
+ENV_HEROKU_RELEASE_VERSION=$(cat "$ENV_DIR/HEROKU_RELEASE_VERSION" 2>/dev/null || date +%s)
 
-if [ -z "$ENV_S3_BUCKET_NAME" ]; then
-  echo "-----> ⏭️  Skipping AWS sync. S3_BUCKET_NAME is not set"
-  exit 0
-fi
+echo "-----> ⌛️  Building frontend"
 
-if [ -z "$ENV_DISTRIBUTION_ID" ]; then
-  echo "-----> ⏭️  Skipping AWS sync. DISTRIBUTION_ID is not set"
-  exit 0
-fi
+cd "$BUILD_DIR"
+yarn build:apps
+cd "$ROOT_DIR"
 
-if [ -z "$ENV_AWS_ACCESS_KEY_ID" ]; then
-  echo "-----> ⏭️  Skipping AWS sync. AWS_ACCESS_KEY_ID is not set"
-  exit 0
-fi
+echo "-----> ✅  Frontend built successfully"
 
-if [ -z "$ENV_AWS_SECRET_ACCESS_KEY" ]; then
-  echo "-----> ⏭️  Skipping AWS sync. AWS_SECRET_ACCESS_KEY is not set"
-  exit 0
-fi
+if [ -n "$ENV_ROLE_ARN" ] && [ -n "$ENV_S3_BUCKET_NAME" ] && [ -n "$ENV_DISTRIBUTION_ID" ] && \
+   [ -n "$ENV_AWS_ACCESS_KEY_ID" ] && [ -n "$ENV_AWS_SECRET_ACCESS_KEY" ] && \
+   [ -n "$ENV_HEROKU_APP_NAME" ] && [ -n "$ENV_HEROKU_RELEASE_VERSION" ]; then
+  echo "-----> ⌛️  Setting env variables"
```

Comment:
```suggestion
echo "-----> ⌛️Starting Cloudfront upload and invalidation process"
echo "-----> ⌛️ Setting env variables"
```

Since now the buildpack does more than 1 thing, it doesn't hurt to be clearer in the output :)

Reply:
done 1ed99afef1f9416daeb4794448aeca604d48b19a